examples/pubsub.rs

```diff
@@ -57,39 +59,39 @@ impl subscriber::FutureService for Subscriber {
 }
 
 impl Subscriber {
-    fn listen(id: u32) -> SocketAddr {
+    fn listen(id: u32, options: server::Options) -> SocketAddr {
         Subscriber { id: id }
-            .listen("localhost:0".first_socket_addr(),
-                    server::Options::default())
-            .wait()
+            .listen("localhost:0".first_socket_addr(), options)
             .unwrap()
     }
 }
 
 #[derive(Clone, Debug)]
 struct Publisher {
-    clients: Arc<Mutex<HashMap<u32, subscriber::FutureClient>>>,
+    clients: Rc<RefCell<HashMap<u32, subscriber::FutureClient>>>,
```

Comment:
Arguably this could actually be an `UnsafeCell`, which would make this explicit in the form of an `unsafe` where we document exactly why we believe there can be no other writers.